### ISSUE TITLE
Builds broken when upgrading to Gradle 6.0.1

### ISSUE THREAD

**Issue by chanseokoh** [GoogleContainerTools/jib#2505]
As revealed in PR #2504.

```
* Where:
Build file '/home/travis/build/gliptak/jib/jib-gradle-plugin/build.gradle' line: 2
* What went wrong:
An exception occurred applying plugin request [id: 'java-gradle-plugin']
> Failed to apply plugin [class 'org.gradle.api.plugins.JavaLibraryPlugin']
   > Cannot create variant 'classes' after configuration ':jib-gradle-plugin:apiElements' has been resolved
```

I think `:apiElements` is from the [Java Library plugin](https://docs.gradle.org/current/userguide/java_library_plugin.html) (`java-library`). And the error seems to occur when applying `java-gradle-plugin`, whose [doc](https://docs.gradle.org/current/userguide/java_gradle_plugin.html) says "[i]t automatically applies the Java Library plugin." The error message does seem to indicate that `java-gradle-plugin` is automatically applying `java-library`. But what's wrong? I'm not a Gradle expert, so not sure what's going on.

_Originally posted by @chanseokoh in https://github.com/GoogleContainerTools/jib/pull/2504#issuecomment-636927432_

**Comment by loosebazooka** [comment]
I wonder if this is something that happened in 6.0 onwards.

I checked the release notes at: https://docs.gradle.org/6.0/release-notes.html
and I think this was merged in for 6.0: https://github.com/gradle/gradle/issues/1494

I think the fix for us is to just apply the java-library plugin earlier in the build (similar to the extension libs?)

**Comment by chanseokoh** [comment]
What I remember back then is that technically it's possible to upgrade our Gradle build version to 6.0.1, but I then our test code becomes incompatible with Gradle < 6 (if my memory serves me right). Since we are still targeting Gradle 5.1+ users, so I think we have to stay on Gradle 5.x at the moment. But hopefully we can advance the baseline to Gradle 6 soon.

**Comment by dubinsky** [comment]
Didn't work for me so far, but thanks for the tip.

**Comment by chanseokoh** [comment]
Fixed by #2952. This repo is now building with Gradle 6.
